evaluation_script/main.py

```diff
@@ -1,5 +1,48 @@
-import os
-import gzip
+from .utils import get_file_extension
+from .utils import open_file
+from .utils import are_all_base_metrics_zero
+from .utils import calculate_precision
+from .utils import calculate_recall
+from .utils import calculate_fpr
+from .utils import calculate_f1
+from .utils import BaseMetrics
+from .utils import Metrics
+
+query_level_base_metrics = {}
+query_level_metrics = {}
+documents_with_ground_truth = set()
+
+"""
+    This function calculates query level metrics from query level base metrics
+"""
+def calculate_query_level_metrics():
+    qa_precision = 0
+    qa_recall = 0
+    qa_fpr = 0
+    qa_f1 = 0
+
+    for query_id in query_level_base_metrics:
+        base_metrics = query_level_base_metrics[query_id]
+        if not are_all_base_metrics_zero(base_metrics.tp, base_metrics.tn, base_metrics.fp, base_metrics.fn):
```

Comment:
Should this occur inside of the `calculate_xyz` routines?

Reply:
No, because `calculate_xyz` will check only for that particular metric but I need to check if they are all `0` at once. If some of them are `0`, that is ok. I need this handling only when all of these become `0`. If I wont put it then `F1` gets a value of `1` because if `tp`, `fp` and `fn` are all `0`, precision and recall get the value of `1`. I dont want the `F1` to be `1` in such scenario. This check helps me prevent that case.